defold-spine/commonsrc/vertices.cpp

```diff
@@ -111,36 +126,109 @@ void GetSkeletonBounds(const spSkeleton* skeleton, SpineModelBounds& bounds)
             }
         }
     }
+}
 
+static void CalcAndAddVertexBufferAttachment(spAttachment* attachment, uint32_t* out_count, uint32_t* out_max_triangle_count)
+{
+    if (attachment->type == SP_ATTACHMENT_REGION)
+    {
+        *out_count += 6;
+    }
+    else if (attachment->type == SP_ATTACHMENT_MESH)
+    {
+        spMeshAttachment* mesh     = (spMeshAttachment*)attachment;
+        uint32_t num_tri_vertices  = SUPER(mesh)->worldVerticesLength;
+        *out_count                += (uint32_t)mesh->trianglesCount; // It's a list of indices, where each 3-tuple define a triangle
+
+        if (num_tri_vertices > *out_max_triangle_count)
+        {
+            *out_max_triangle_count = num_tri_vertices;
+        }
+    }
 }
 
-uint32_t CalcVertexBufferSize(const spSkeleton* skeleton, uint32_t* out_max_triangle_count)
+static void CalcAndAddVertexBufferAttachmentByClipper(dmArray<float>& scratch, spSlot* slot, spAttachment* attachment, spSkeletonClipping* skeleton_clipper, uint32_t* out_count, uint32_t* out_max_triangle_count)
```

Comment:
I wonder how slow this is (I don't have any good data to test with) - perhaps we should pass in the scratch buffer to the GenerateVertexData function and resize/reallocate as needed? We need to at least do this once for generating the vertex buffer size, but perhaps the second time could be avoided 🤔